src/greeter/GreeterApp.cpp

```diff
@@ -134,6 +155,39 @@ namespace SDDM {
         connect(this, &GreeterApp::primaryScreenChanged, this, [this](QScreen *) {
             activatePrimary();
         });
+
+        // create filewatcher for adminInfo-file
+        QString adminInfoPath = SDDM::mainConfig.AdminInfoPath.get();
+
+        // check if config has been set
+        if (!adminInfoPath.isEmpty()) {
+            // check for theme-compatibility
+            if (m_metadata->adminInfoSupport()) {
+                m_fileAgent = new QFileSystemWatcher(this);
+
+                // connect signals
+                connect(m_fileAgent, &QFileSystemWatcher::fileChanged, this, &GreeterApp::fileChanged);
```

Comment:
No, you are right. It does not work in this specific scenario.
But keep in mind that your solution (if I understood it correctly) won't solve it either:

1. our code already "monitors" the directory  since `absoluteFilePath()` does the same as `absoluteDir()` and simply returns it in a different format.
 2. directoryChanged signals will only be sent when a file gets created or deleted, but not if it gets modified.

So the solution for this bug is probably way more complicated than necessary?